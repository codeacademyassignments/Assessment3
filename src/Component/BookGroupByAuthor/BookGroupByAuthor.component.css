.cardBoundBox{
  display: flex;
    /* flex-wrap: wrap; */
    flex-direction: column;
    padding-bottom: 34px;
    min-width: 350px;
    justify-content: space-around;
    align-items: center;
    border: 6px solid orange;
    border-radius: 47px;
    margin: 3% 0%;

}
.cardsByAuthor{
  display: flex;
    flex-wrap: wrap;
    justify-content: space-around;
    min-width: 350px;
}
.authorName{
  color: orange;
  font-size: 29px;
  background-color: white;
  border-radius: 7%;
  padding: 9px 50px;
  z-index: 30;
  font-weight: bold;
  position: relative;
  bottom: 28px;
}