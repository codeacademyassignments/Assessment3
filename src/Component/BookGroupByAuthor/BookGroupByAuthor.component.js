import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Book from '../Book/Book.component';
import './BookGroupByAuthor.component.css';

// import ReactDOM from 'react-dom';
// import RawData from '../../RawData/RawData';

class BookGroupByAuthor extends Component {
  createBookCards = (books) => {
    const bookCards = [];
    books.forEach((book, index) => {
      bookCards.push(<Book key={`${book.Name}`} bookId={book.id} liked={book.liked} rating={book.rating} title={book.Name} index={index} />);
    });
    return bookCards;
  }

  render() {
    const { books, author } = this.props;
    return (
      <section className="cardBoundBox">
        <div className="authorName">{author}</div>
        <section className="cardsByAuthor">
          {this.createBookCards(books)}
        </section>
      </section>

    );
  }
}

BookGroupByAuthor.propTypes = {
  books: PropTypes.arrayOf(
    PropTypes.shape({
      Name: PropTypes.string.isRequired,
      id: PropTypes.number.isRequired,
      rating: PropTypes.float,
      Author: PropTypes.string.isRequired,
    }),
  ).isRequired,
  author: PropTypes.string.isRequired,
};


export default BookGroupByAuthor;
