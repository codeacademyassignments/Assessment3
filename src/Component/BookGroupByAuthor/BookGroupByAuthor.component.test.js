import React from 'react';
import renderer from 'react-test-renderer';
import BookGroupByAuthor from './BookGroupByAuthor.component';
import TestData from '../../TestData/TestData';

const books = TestData['J K Rowling'];
describe('Section', () => {
  it('snapshot should match', () => {
    const tree = renderer.create(<BookGroupByAuthor books={books} author="J K Rowling" />);
    expect(tree).toMatchSnapshot();
  });
});
