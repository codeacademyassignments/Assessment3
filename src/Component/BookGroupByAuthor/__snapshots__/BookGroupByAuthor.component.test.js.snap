// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshot should match 1`] = `
<section
  className="cardBoundBox"
>
  <div
    className="authorName"
  >
    J K Rowling
  </div>
  <section
    className="cardsByAuthor"
  >
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Sorcerers Stone (Harry Potter, #1)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.45
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Chamber of Secrets (Harry Potter, #2)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.38
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Prisoner of Azkaban (Harry Potter, #3)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.54
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Goblet of Fire (Harry Potter, #4)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.53
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Order of the Phoenix (Harry Potter, #5)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.47
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Half-Blood Prince (Harry Potter, #6)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.54
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
    <article
      className="book"
    >
      <div
        className="bookImage"
        style={
          Object {
            "backgroundImage": "url(cover_image.jpeg)",
          }
        }
      />
      <p
        className="bookTitle"
      >
        Harry Potter and the Deathly Hallows (Harry Potter, #7)
      </p>
      <div
        className="cardFooter"
      >
        <div>
          <p
            className="rating"
          >
            4.62
          </p>
        </div>
        <div
          className="heartImage"
        >
          <img
            alt="heart-black"
            className="heart"
            onClick={[Function]}
            src={
              Object {
                "ReactComponent": [Function],
                "__esModule": true,
                "default": "heart-white.svg",
              }
            }
          />
        </div>
      </div>
    </article>
  </section>
</section>
`;
